pkg/controller/common.go

```diff
@@ -158,60 +166,66 @@ func (c *SharedState) getPodsFromVolume(volumeName string) ([]string, error) {
-		volumes = []string{}
-		for _, pod := range pods {
-			podVolumes, err := c.getVolumesFromPod(pod)
-			if err != nil {
-				return nil, err
-			}
-			volumes = append(volumes, podVolumes...)
-		}
-	}
-
-	numReplica := 0
-	for _, volume := range volumes {
-		var azVolume v1alpha1.AzVolume
-		if err := azr.getClient().Get(ctx, types.NamespacedName{Namespace: azr.getNamespace(), Name: volume}, &azVolume); err != nil {
+func (c *SharedState) getVolumesForPods(pods ...string) ([]string, error) {
+	volumes := []string{}
+	for _, pod := range pods {
+		podVolumes, err := c.getVolumesFromPod(pod)
+		if err != nil {
 			return nil, err
 		}
-		numReplica = max(numReplica, azVolume.Spec.MaxMountReplicaCount)
+		volumes = append(volumes, podVolumes...)
 	}
+	return volumes, nil
+}
 
+func getRankedNodesForReplicaAttachments(ctx context.Context, azr azReconciler, volumes []string) ([]string, error) {
+	klog.V(5).Info("Getting ranked list of nodes for creating AzVolumeAttachments")
+	numReplica := 0
 	nodes := []string{}
 	nodeScores := map[string]int{}
 	primaryNode := ""
+
 	for _, volume := range volumes {
-		volReq, err := createLabelRequirements(azureutils.VolumeNameLabel, volume)
+		azVolume, err := azureutils.GetAzVolume(ctx, azr.getClient(), azr.getAzClient(), volume, consts.AzureDiskCrdNamespace, true)
 		if err != nil {
+			klog.V(5).Info(fmt.Sprintf("AzVolume for volume %s is not found.", volume))
 			return nil, err
 		}
-		labelSelector := labels.NewSelector().Add(*volReq)
-		azVolumeAttachments := &v1alpha1.AzVolumeAttachmentList{}
-		err = azr.getClient().List(ctx, azVolumeAttachments, &client.ListOptions{LabelSelector: labelSelector})
+
+		numReplica = max(numReplica, azVolume.Spec.MaxMountReplicaCount)
+		klog.V(5).Info(fmt.Sprintf("Number of requested replicas for azvolume %s is: %d. Max replica count is: %d.",
+			volume, numReplica, azVolume.Spec.MaxMountReplicaCount))
+
+		azVolumeAttachments, err := getAzVolumeAttachmentsForVolume(ctx, azr.getClient(), volume, all)
 		if err != nil {
-			return nil, err
+			klog.V(5).Info(fmt.Sprintf("Error listing AzVolumeAttachments for azvolume %s. Error: %v.", volume, err))
 		}
 
-		for _, azVolumeAttachment := range azVolumeAttachments.Items {
+		klog.V(5).Info(fmt.Sprintf("Found %d AzVolumeAttachments for volume %s. AzVolumeAttachments: %v.", len(azVolumeAttachments), volume, azVolumeAttachments))
+		for _, azVolumeAttachment := range azVolumeAttachments {
 			if azVolumeAttachment.Spec.RequestedRole == v1alpha1.PrimaryRole {
+				klog.V(5).Info(fmt.Sprintf("AzVolumeAttachments %s for volume %s is primary on node %s.", azVolumeAttachment.Name, volume, azVolumeAttachment.Spec.NodeName))
 				if primaryNode == "" {
 					primaryNode = azVolumeAttachment.Spec.NodeName
 				}
-				continue
-			}
-			if _, ok := nodeScores[azVolumeAttachment.Spec.NodeName]; !ok {
-				nodeScores[azVolumeAttachment.Spec.NodeName] = 0
-				nodes = append(nodes, azVolumeAttachment.Spec.NodeName)
+			} else {
```

Comment:
np: could be replaced with `if { continue } ...` to keep the style consistent.

Reply:
I think the two styles are pretty much identical in terms of case separation but if you feel so, I am fine with the change :)